storage-access.bs

```diff
@@ -65,53 +69,84 @@ This specification depends on the Infra standard. [[!INFRA]]
+
+1. the [=active document=] of its [=Document/browsing context=]'s [=parent browsing context=] is in a [=first party context=], and
+1. the [=environment settings object/origin=] and [=top-level origin=] of its [=relevant settings object=] are [=same site=] with one another.
+
+A {{Document}} is in a <dfn>third party context</dfn> if it is not in a [=first party context=].
+
+ISSUE(10): If we let nested <{iframe}>s use this API, we may have to revisit these definitions.
+
 <h3 id="ua-state">User Agent state related to storage access</h3>
 
-Each User Agent maintains a <dfn>storage access map</dfn>, an [=ordered map=] whose keys are [=partitioned storage keys=] and whose values are [=storage access flag sets=].
+A <dfn>storage access map</dfn> is a [=map=] whose keys are [=partitioned storage keys=] and whose values are [=storage access flag sets=].
 
-ISSUE: What's the lifetime of the [=storage access map=] and of its entries?
+User Agents maintain a single <dfn>global storage access map</dfn>.
 
-ISSUE(28): Tie the [=storage access map=] to agent clusters.
+ISSUE(2): What is the lifecycle of the [=global storage access map=]? How long do we remember its contents? Firefox and Safari differ here.
 
-A <dfn>partitioned storage key</dfn> is a [=pair=] consisting of a top-level site and a site, both [=sites=].
+ISSUE(5): When do we age out entries in the [=global storage access map=]? See also [Scope of Storage Access](https://github.com/privacycg/storage-access#scope-of-storage-access).
 
-ISSUE: Replace this [=partitioned storage key=] concept with whatever Anne comes up with for [[!STORAGE]].
+Each [=agent cluster=] has a <dfn for="agent cluster">storage access map</dfn>.
 
-A <dfn>storage access flag set</dfn> is a set of zero or more of the following flags, which are used to gate access to client-side storage for |site| when it's a [=third party=] on |top-level site|:
+When an [=agent cluster=] is created, its [=agent cluster/storage access map=] is initialized with a [=map/clone=] of the [=global storage access map=].
 
-: The <dfn for="storage access flag set" id=has-storage-access-flag>has storage access flag</dfn>
-:: When set, this flag indicates |site| has access to its [=first party data=] when it's a [=third party=] on |top-level site|.
-: The <dfn for="storage access flag set" id=was-expressly-denied-storage-access-flag>was expressly denied storage access flag</dfn>
-:: When set, this flag indicates that the user expressly denied |site| access to its [=first party data=] when it's a [=third party=] on |top-level site|.
+To <dfn type="abstract-op">obtain the storage access map</dfn> for a {{Document}} |doc|, run the following steps:
+
+1. If |doc|'s [=Document/browsing context=] is not a [=top-level browsing context=], set |doc| to the [=active document=] of |doc|'s [=Document/browsing context=]'s [=top-level browsing context=].
```

Comment:
Why do we need to go to the top-level here?

Reply:
I suppose we don't. Fixed in 93b9650.